crates/bitwarden-auth/src/identity/login_via_password/login_via_password_impl.rs

```diff
@@ -0,0 +1,592 @@
+use bitwarden_core::key_management::MasterPasswordAuthenticationData;
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+use crate::identity::{
+    LoginClient,
+    api::{request::LoginApiRequest, send_login_request},
+    login_via_password::{PasswordLoginApiRequest, PasswordLoginError, PasswordLoginRequest},
+    models::LoginResponse,
+};
+
+#[cfg_attr(feature = "wasm", wasm_bindgen)]
+#[cfg_attr(feature = "uniffi", uniffi::export)]
+impl LoginClient {
```

Comment:
Style: I believe putting `impl` for a struct into a separate file is not common / does not exist in the rest of the SDK. It should rather live in the `login_login_client` file?

If the reason is that you want to split by login strategy, and have those in separate files, then the structuring still seems wrong, since you do not want to have multiple impl blocks for one struct (and i'm not even sure if it works with bindgen / uniffi).

One pattern here would be to have the impl block in loginClient, and then delegate out to functions that you have in separate files, which is the pattern that the sdk uses for clients.

Reply:
I appreciate this feedback. @dani-garcia and I explicitly talked about this during the initial design sessions that we had, and from what I understood, multiple `impl` blocks were supposed to be fine. If this will cause issues with bindgen / uniffi, then I'm glad to know now. Do you have an example of an impl delegating out to functions in separate files?